docs/IntentReader.md

```diff
@@ -1,62 +1,127 @@
+| task            | The value of the task determines the handler to be called. This is a constant value for a particular purpose. (i.e., for CallMethodHandler, the task is "callMethod, and for GetPubSubStatusHandler, the task is "getPubSubStatus"). | Y                             |
+| params          | The params required for the INTENT                                                                                                                                                                                                   | N - depends on the INTENT     |
+| appType         | Type of app ( eg: "firebolt")                                                                                                                                                                                                        | Y                             |
 ## Pub Sub
-
 ### Background
 
-TODO: add details
+FCA has the capability to send and receive messages by making use of a PubSub (Publish-Subscribe) messaging system. FCA makes use of the messaging system by integrating it as a plugin. Setup to configure and integrate a messaging system is outlined [here](../plugins/PubSub.md).
+
+Once the PubSub plugin is integrated into FCA, it can send or receive messages by publishing and listening to various topics.
+We can send commands/intents to FCA, which will be parsed and formatted by using various handlers.
 
+You can find the code for handling different types of PubSub requests in the [intentReaderHandlers directory.](./intentReaderHandlers)
+For pubSub scenarios, initially pubSub client communication is established, and then after getting the client response message, FCA processes the incoming intents, and corresponding handlers are called as per the task in the intent.
 ### Setup
 
-TODO: add details - link to plugins docs for pub sub
+The PubSub plugin has to be integrated with FCA to call IntentReader with PubSub communication. Setup details [here.](../plugins/PubSub.md)
 
 ### Usage
 
-TODO: add details
+With the configuration and setup in place, we can send messages to FCA through any pubsub mechanism like Websocket, etc. First, we have to initialise a pubsub connection, then publish and subscribe topics to send or receive messages from FCA.
+We can perform any action or task with intentReaders and with the help of multiple handlers. For performing a particular task, give the required task name in the "task" field and the corresponding task parameters required to perform the task.
+For example:
+To invoke any APIs, the task will be "callMethod", and the corresponding methodName, methodParams, etc. should be passed.
+Here, pubSub communication status is established first, and then intent is passed.
 
```

Comment:
Give them as bullet points instead of a paragraph . 
eg for sending the intents via Pubsub , below steps have to be followed.

- The sender (Can be a framework/server or an application) needs to have Pubsub plugin integrated into it. 
- Once the Pubsub plugin is integrated into the sender, it has to subscribe to the topic FCA will be publishing. Server can then publish the intent to the same topic which FCA has subscribed to.